components/script/dom/htmldialogelement.rs

```diff
@@ -70,6 +70,33 @@ impl HTMLDialogElementMethods for HTMLDialogElement {
         *self.return_value.borrow_mut() = return_value;
     }
 
+    // https://html.spec.whatwg.org/multipage/#dom-dialog-show
+    fn Show(&self) {
```

Comment:
This partially implements the spec, most of the TODOs are related to popover API interactions which arent' supported yet, or the focus steps which again aren't implemented yet (also not implemented for close afaict)